TST: test_bounds_infeasible_2 failing on wheels repo cron jobs

This failure is showing up in [weekly cron jobs](https://travis-ci.org/github/MacPython/scipy-wheels/builds/708193097) for `scipy-wheels` repo (`master` branch builds) for the last two weeks, in both MacOS and Linux builds & for both 64- and 32-bit cases.

Sample traceback:

```
=================================== FAILURES ===================================
____________ TestLinprogSimplexNoPresolve.test_bounds_infeasible_2 _____________
[gw1] linux -- Python 3.6.8 /venv/bin/python
/venv/lib/python3.6/site-packages/scipy/optimize/tests/test_linprog.py:435: in test_bounds_infeasible_2
    assert_raises(RuntimeWarning, g, c, bounds=bounds_1)
        bounds_1   = [(1, 2), (inf, inf), (3, 4)]
        bounds_2   = [(1, 2), (-inf, -inf), (3, 4)]
        c          = [1, 2, 3]
        do_presolve = False
        g          = <function LinprogCommonTests.test_bounds_infeasible_2.<locals>.g at 0xe0682c44>
        self       = <scipy.optimize.tests.test_linprog.TestLinprogSimplexNoPresolve object at 0xe06814ac>
        simplex_without_presolve = True
/venv/lib/python3.6/site-packages/scipy/optimize/tests/test_linprog.py:432: in g
    res = linprog(c, bounds=bounds, method=self.method, options=self.options)
        bounds     = [(1, 2), (inf, inf), (3, 4)]
        c          = [1, 2, 3]
        self       = <scipy.optimize.tests.test_linprog.TestLinprogSimplexNoPresolve object at 0xe06814ac>
/venv/lib/python3.6/site-packages/scipy/optimize/_linprog.py:545: in linprog
    postsolve_args=postsolve_args, **solver_options)
        A          = array([[1., 0., 0., 1., 0.],
       [0., 0., 1., 0., 1.]])
        A_eq       = None
        A_ub       = None
        C          = 1
        b          = array([nan, nan])
        b_eq       = None
        b_scale    = 1
        b_ub       = None
        bounds     = [(1, 2), (inf, inf), (3, 4)]
        c          = array([1., 2., 3., 0., 0.])
        c0         = inf
        callback   = None
        complete   = False
        iteration  = 0
        lp         = _LPProblem(c=array([1., 2., 3.]), A_ub=array([], shape=(0, 3), dtype=float64), b_ub=array([], dtype=float64), A_eq=arr...ype=float64), b_eq=array([], dtype=float64), bounds=array([[ 1.,  2.],
       [inf, inf],
       [ 3.,  4.]]), x0=None)
        lp_o       = _LPProblem(c=array([1., 2., 3.]), A_ub=array([], shape=(0, 3), dtype=float64), b_ub=array([], dtype=float64), A_eq=arr...ype=float64), b_eq=array([], dtype=float64), bounds=array([[ 1.,  2.],
       [inf, inf],
       [ 3.,  4.]]), x0=None)
        meth       = 'simplex'
        method     = 'simplex'
        options    = {'presolve': False}
        postsolve_args = (_LPProblem(c=array([1., 2., 3.]), A_ub=array([], shape=(0, 3), dtype=float64), b_ub=array([], dtype=float64), A_eq=ar...), b_eq=array([], dtype=float64), bounds=array([[ 1.,  2.],
       [inf, inf],
       [ 3.,  4.]]), x0=None), [], 1, 1)
        solver_options = {}
        tol        = 1e-09
        undo       = []
        x0         = None
/venv/lib/python3.6/site-packages/scipy/optimize/_linprog_simplex.py:624: in _linprog_simplex
    bland=bland
        A          = array([[1., 0., 0., 1., 0.],
       [0., 0., 1., 0., 1.]])
        T          = array([[ 1.,  0.,  0.,  1.,  0.,  1.,  0., nan],
       [ 0.,  0.,  1.,  0.,  1.,  0.,  1., nan],
       [ 1.,  2.,  3.,  0.,  0.,  0.,  0., inf],
       [-1., -0., -1., -1., -1.,  0.,  0., nan]])
        av         = array([5, 6])
        b          = array([nan, nan])
        basis      = array([5, 6])
        bland      = False
        c          = array([1., 2., 3., 0., 0.])
        c0         = inf
        callback   = None
        disp       = False
        is_negative_constraint = array([False, False])
        m          = 5
        maxiter    = 1000
        messages   = {0: 'Optimization terminated successfully.', 1: 'Iteration limit reached.', 2: 'Optimization failed. Unable to find a feasible starting point.', 3: 'Optimization failed. The problem appears to be unbounded.', ...}
        n          = 2
        postsolve_args = (_LPProblem(c=array([1., 2., 3.]), A_ub=array([], shape=(0, 3), dtype=float64), b_ub=array([], dtype=float64), A_eq=ar...), b_eq=array([], dtype=float64), bounds=array([[ 1.,  2.],
       [inf, inf],
       [ 3.,  4.]]), x0=None), [], 1, 1)
        row_constraints = array([[ 1.,  0.,  0.,  1.,  0.,  1.,  0., nan],
       [ 0.,  0.,  1.,  0.,  1.,  0.,  1., nan]])
        row_objective = array([ 1.,  2.,  3.,  0.,  0.,  0.,  0., inf])
        row_pseudo_objective = array([-1., -0., -1., -1., -1.,  0.,  0., nan])
        status     = 0
        tol        = 1e-09
        unknown_options = {}
/venv/lib/python3.6/site-packages/scipy/optimize/_linprog_simplex.py:401: in _solve_simplex
    pivrow_found, pivrow = _pivot_row(T, basis, pivcol, phase, tol, bland)
        T          = array([[ 1.,  0.,  0.,  1.,  0.,  1.,  0., nan],
       [ 0.,  0.,  1.,  0.,  1.,  0.,  1., nan],
       [ 1.,  2.,  3.,  0.,  0.,  0.,  0., inf],
       [-1., -0., -1., -1., -1.,  0.,  0., nan]])
        basis      = array([5, 6])
        bland      = False
        callback   = None
        complete   = False
        m          = 6
        maxiter    = 1000
        message    = ''
        n          = 2
        nit        = 0
        nit0       = 0
        phase      = 1
        pivcol     = 0
        pivcol_found = True
        postsolve_args = (_LPProblem(c=array([1., 2., 3.]), A_ub=array([], shape=(0, 3), dtype=float64), b_ub=array([], dtype=float64), A_eq=ar...), b_eq=array([], dtype=float64), bounds=array([[ 1.,  2.],
       [inf, inf],
       [ 3.,  4.]]), x0=None), [], 1, 1)
        solution   = array([  0.,   0.,   2.,   0.,   0.,   0., -11.])
        status     = 0
        tol        = 1e-09
/venv/lib/python3.6/site-packages/scipy/optimize/_linprog_simplex.py:166: in _pivot_row
    return True, min_rows[0]
E   IndexError: index 0 is out of bounds for axis 0 with size 0
        T          = array([[ 1.,  0.,  0.,  1.,  0.,  1.,  0., nan],
       [ 0.,  0.,  1.,  0.,  1.,  0.,  1., nan],
       [ 1.,  2.,  3.,  0.,  0.,  0.,  0., inf],
       [-1., -0., -1., -1., -1.,  0.,  0., nan]])
        basis      = array([5, 6])
        bland      = False
        k          = 2
        ma         = masked_array(data=[1.0, --],
             mask=[False,  True],
       fill_value=1e+20)
        mb         = masked_array(data=[nan, --],
             mask=[False,  True],
       fill_value=1e+20)
        min_rows   = array([], dtype=int32)
        phase      = 1
        pivcol     = 0
        q          = masked_array(data=[--, --],
             mask=[ True,  True],
       fill_value=1e+20,
            dtype=float64)
        tol        = 1e-09
=============================== warnings summary ===============================
/venv/lib/python3.6/site-packages/scipy/optimize/_linprog_util.py:1156
  /venv/lib/python3.6/site-packages/scipy/optimize/_linprog_util.py:1156: RuntimeWarning: invalid value encountered in multiply
    b -= (A[:, i_shift] * lb_shift).sum(axis=1)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
```

--- a/scipy/optimize/tests/test_linprog.py
+++ b/scipy/optimize/tests/test_linprog.py
@@ -432,8 +432,13 @@ class LinprogCommonTests(object):
                 res = linprog(c, bounds=bounds, method=self.method, options=self.options)
                 return res
 
-            assert_raises(RuntimeWarning, g, c, bounds=bounds_1)
-            assert_raises(RuntimeWarning, g, c, bounds=bounds_2)
+            with pytest.warns(RuntimeWarning):
+                with pytest.raises(IndexError):
+                    g(c, bounds=bounds_1)
+
+            with pytest.warns(RuntimeWarning):
+                with pytest.raises(IndexError):
+                    g(c, bounds=bounds_2)
         else:
             res = linprog(c=c, bounds=bounds_1, method=self.method, options=self.options)
             _assert_infeasible(res)

